src/validation/water_uptake/modal_aero_water_uptake_dr_col.cpp

```diff
@@ -0,0 +1,124 @@
+    using View1DHost = typename HostType::view_1d<Real>;
+
+    constexpr int nvars = ndrop::nvars;
+    constexpr int pver = ndrop::pver;
+    constexpr int ntot_amode = AeroConfig::num_modes();
+
+    View2D state_q("state_q", pver, nvars);
+    mam4::validation::convert_1d_vector_to_2d_view_device(state_q_db, state_q);
+
+    ColumnView temperature;
+    temperature = haero::testing::create_column_view(pver);
+    auto temperature_host = View1DHost((Real *)temperature_db.data(), pver);
+
+    ColumnView pmid;
+    pmid = haero::testing::create_column_view(pver);
+    auto pmid_host = View1DHost((Real *)pmid_db.data(), pver);
+
+    ColumnView cldn;
+    cldn = haero::testing::create_column_view(pver);
+    auto cldn_host = View1DHost((Real *)cldn_db.data(), pver);
+
+    Kokkos::deep_copy(temperature, temperature_host);
+    Kokkos::deep_copy(pmid, pmid_host);
+    Kokkos::deep_copy(cldn, cldn_host);
+
+    View2D dgncur_a("dgncur_a_db", pver, ntot_amode);
+    View2D dgncur_awet("dgncur_awet", pver, ntot_amode);
+    View2D qaerwat("qaerwat", pver, ntot_amode);
+    mam4::validation::convert_1d_vector_to_2d_view_device(dgncur_a_db,
+                                                          dgncur_a);
+
+    mam4::validation::convert_1d_vector_to_2d_view_device(dgncur_awet_db,
+                                                          dgncur_awet);
+
+    mam4::validation::convert_1d_vector_to_2d_view_device(qaerwat_db, qaerwat);
+
+    {
+
+      const int top_lev = 1;
+      auto team_policy = ThreadTeamPolicy(1, 1u);
+      Kokkos::parallel_for(
+          team_policy, KOKKOS_LAMBDA(const ThreadTeam &team) {
+            for (int kk = top_lev; kk < pver; ++kk) {
+
+              int nspec_amode[AeroConfig::num_modes()];
+              int lspectype_amode[water_uptake::maxd_aspectype]
+                                 [AeroConfig::num_modes()];
+              Real specdens_amode[water_uptake::maxd_aspectype];
+              Real spechygro[water_uptake::maxd_aspectype];
+
+              water_uptake::get_e3sm_parameters(nspec_amode, lspectype_amode,
+                                                specdens_amode, spechygro);
+
+              const auto state_q_kk =
+                  Kokkos::subview(state_q, kk, Kokkos::ALL());
+              const auto dgnumdry_m_kk =
+                  Kokkos::subview(dgncur_a, kk, Kokkos::ALL());
+              const auto dgnumwet_m_kk =
+                  Kokkos::subview(dgncur_awet, kk, Kokkos::ALL());
+              const auto qaerwat_m_kk =
+                  Kokkos::subview(qaerwat, kk, Kokkos::ALL());
+              mam4::water_uptake::modal_aero_water_uptake_dr(
+                  nspec_amode, specdens_amode, spechygro, lspectype_amode,
+                  state_q_kk.data(), temperature(kk), pmid(kk), cldn(kk),
+                  dgnumdry_m_kk.data(), dgnumwet_m_kk.data(),
+                  qaerwat_m_kk.data());
+
+            } // kk
+          });
+
+      // placeholder
```

Comment:
is anything else needed here?

Reply:
Anything is needed here. I deleted this comment.